scripts/release.sh

```diff
@@ -0,0 +1,291 @@
+#!/bin/bash
+
+# George Miller
+# 05-17-2022
+# If you don't have permission to run, try: chmod u+x release.sh
+# numbering follows this document: https://github.com/smartdevicelink/sdl_ios/wiki/Release-Steps
+# The numbering does restart halfway because the document does.
+
+# a utility function for prompting the user Y/N
+# takes in a string promt for the input
+# returns 1 for yes/true or 0 for no/false
+prompt_user() {
+    user_input="g"
+    echo
+    echo $1" (Y/N)?"
+    read user_input
+    while [[ ! $user_input == [YyNn] ]]; do
+        echo $1" (Y/N)?"
+        read user_input
+    done
+    if [[ ! $user_input == [Nn] ]]; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+# TODO - phase 4 - github cli "gh" needs to be installed before we can use those commands.  We could automate this or at least check if GH is installed.
+#  gh commands are commented for now.
+
+# script start
+echo 
+echo "Starting SDL release script..."
+
+# If we are running from the scripts directory, we want to pop back to the project root to do everything.
+if [[ $PWD == *"scripts" ]]; then
+    cd ..
+fi
+# If, for some reason, we are not now in the correct working directory, exit
+if [[ $PWD != *"sdl_ios" ]]; then
+    echo "Please run this from the sdl_ios project root or the sdl_ios/scripts directory"
+    exit 0
+fi
+
+# setup branch variables
+develop_branch="develop"
+main_branch="master"
+
+
+#DEBUG - copy out framework script
+#this is necessary because the branch that these scripts are being developed on has not yet been merged into develop.  This will not be necessary in production.
+cp scripts/create_framework.sh ../create_framework.sh
+
+#stash any local changes to avoid errors during checkout
+git status
+prompt_user "Would you like to stash any local changes?"
+if [[ $? == 1 ]]; then
+    # stash local changes to prevent issues with checkout
+    git stash
+    echo "use \"git stash pop\" when this script is complete to restore your changes"
+else
+    # dump local changes to prevent issues with checkout
+    git reset --hard
+fi
+
+# Checkout develop
+# we need to checkout the branch before we start modifying files.
+echo
+echo "Checking out $develop_branch"
+git checkout $develop_branch
+
+#DEBUG move framework script back in
+#like above, this will not be required after the release script branch is merged into develop
```

Comment:
Please be consistent in comments. One space after `#`, start with a capital letter, one space after a period. Please go through your comments and fix based on that style.

Reply:
Why does the double space bother you?  I admit it is a bad habit left over from when I learned to type as a kid, but most people would not have noticed.

Reply:
Comments are adjusted.